### PR TITLE
🐛 catalog: fixes access-rights to `get` and `update` services

### PR DESCRIPTION
**Pull request by pcrespov**
## What do these changes do?

 🐛 enforces access-rights for `get` and `update` on the catalog service

## Related issue/s

- Part of https://github.com/ITISFoundation/osparc-simcore/issues/6201

## How to test

```
make devenv
cd services/catalog
make install-dev
pytest -v -k test_rpc_get_service_access_rights tests
```

## Dev-ops checklist
None

### CODE REVIEW

**Review by GitHK — APPROVED** [review]
👍  very nice

File: services/catalog/src/simcore_service_catalog/db/repositories/services.py [hunk 1]
```diff
@@ -243,8 +244,8 @@ async def create_or_update_service(
         return created_service
 
     async def update_service(
-        self, patched_service: ServiceMetaDataAtDB
-    ) -> ServiceMetaDataAtDB:
+        self, patched_service: ServiceMetaDataAtDB, *, returning: bool = False
```
Comment (sanderegg) [inline services/catalog/src/simcore_service_catalog/db/repositories/services.py:247]: Question: I always ask myself this. What is the point of having a default parameter if we control the whole code?
1. Are there a lot of locations where we use it and it's painful to change?
2. Does it change something in terms of performance to return the service after the update?

File: services/catalog/src/simcore_service_catalog/services/services_api.py [hunk 2]
```diff
@@ -267,17 +274,17 @@ async def update_service(
         await repo.upsert_service_access_rights(new_access_rights)
 
         # then delete the ones that were removed
-        remove_gids = [gid for gid in before_gids if gid not in update.access_rights]
-        delete_access_rights = [
+        rm_access_rights = [
```
Comment (sanderegg) [inline services/catalog/src/simcore_service_catalog/services/services_api.py:277]:
```suggestion
        removed_access_rights = [
```

File: services/catalog/src/simcore_service_catalog/services/services_api.py [hunk 3]
```diff
@@ -267,17 +274,17 @@ async def update_service(
...
                 version=service_version,
                 gid=gid,
                 product_name=product_name,
             )
-            for gid in remove_gids
+            for gid in previous_gids
+            if gid not in update.access_rights
         ]
-        await repo.delete_service_access_rights(delete_access_rights)
+        await repo.delete_service_access_rights(rm_access_rights)
```
Comment (sanderegg) [inline services/catalog/src/simcore_service_catalog/services/services_api.py:287]:
```suggestion
        await repo.delete_service_access_rights(removed_access_rights)
```
Please let's not spare 5 characters...

**Review by sanderegg — APPROVED** [review]
nice. thanks!

**Pull request merged**